addon/components/ya-input/component.js

```diff
@@ -3,89 +3,86 @@ import layout from '../../templates/ya-input';
 
 const {
   Component,
-  addObserver,
-  getProperties,
+  defineProperty,
   isEmpty,
-  observer,
   on,
+  get,
+  set,
+  getWithDefault,
   computed,
-  run
+  String: { dasherize },
+  computed: { oneWay },
+  A: emberArray
 } = Ember;
 
-const { next: runNext } = run;
-
-const {
-  oneWay
-} = computed;
-
-const get = Ember.get;
-const set = Ember.set;
+function defaultCPSetter(_key, newValue) {
+  return newValue;
+}
 
 export default Component.extend({
-  layout: layout,
+  layout,
+  type: 'text',
   classNameBindings: ['validClass', 'type'],
-
-  field: '',
-  canShowErrors: false,
+  focusedOnce: false,
   hasLostFocus: false,
+  initialFocus: false,
```

Comment:
I think booleans are a lot more expressive when prepended by `is` or `has`.

For example, I'm not sure by the name of this property, what it's keeping track of. Is it keeping track of something having been focused on at a point in the past? If so, `hasReceivedFocus` would be more meaningful IMO.